chainer/functions/array/moveaxis.py

```diff
@@ -0,0 +1,115 @@
+import six
+
+from chainer import cuda
+from chainer import function_node
+from chainer.utils import type_check
+
+
+def _normalize_axis_tuple(axis, ndim, xp):
+    if xp.isscalar(axis):
+        axis = (axis,)
+
+    ret = []
+    for ax in axis:
+        ret.append(ax % ndim)
+    return ret
+
+
+def _moveaxis(a, source, destination, xp):
+    if hasattr(xp, 'moveaxis'):
+        return xp.moveaxis(a, source, destination)
+    source = _normalize_axis_tuple(source, a.ndim, xp)
+    destination = _normalize_axis_tuple(destination, a.ndim, xp)
```

Comment:
How about adding a length check between `source` and `destination` after this line and removing all checks in `__init__` of `Moveaxis` class? Like numpy does it here: https://github.com/numpy/numpy/blob/v1.13.0/numpy/core/numeric.py#L1603-L1605